jane/doc/extensions/_04-modes/intro.md

```diff
@@ -183,5 +183,64 @@ they are once.
 More documentation on uniqueness and linearity is available
 [here](../../uniqueness/intro/).
 
-<!-- CR ccasinghino: Sections needed for statefulness, visibility, and yielding -->
+## Future modes: Yielding
 
+|----------------|
+| yielding       |
+| `|`            |
+| **unyielding** |
+{: .table }
+
+Yielding is a future axis that tracks whether a function is permitted to perform
+effects that will be handled in its parent stack. See [the OCaml Manual entry
+for effect handlers](https://ocaml.org/manual/5.3/effects.html).
+
+Yielding has different defaults depending on the locality axis: *global* values are
+defaulted to *unyielding*, while *local* values are defaulted to *yielding*.
+More documentation on mode implications is available [here](../_05-kinds/syntax.md).
+
+Yielding is irrelevant for types that do not contain functions, and values of such types
+*mode cross* on the yielding axis; they may be used as unyielding even
+when they are yielding.
+
+# Past modes: Visibility
+
+|----------------|
+| immutable      |
+| `|`            |
+| read           |
+| `|`            |
+| **read_write** |
+{: .table}
+
+Visibility is a past axis that controls access to mutable portions of values.
+It's similar to contention: the typechecker forbids accessing mutable fields of values
+with *immutable* visiblity, and forbids writing to mutable fields of values
+with *read* visibility. Unlike contention, atomic access to mutable parts of *immutable*
+values is still disallowed.
+
+Visibility is irrelevant for types that are deeply immutable. Values of such
```

Comment:
This sounds like what's said about contention, but that's slightly wrong, because types such that all their mutable fields are atomic cross contention.

Reply:
Is it better now?

Reply:
No, I don't see a change here. I think the fix is to actually change the doc for contention.

Reply:
Added a line about atomic accesses still being permitted to the contention docs.

Reply:
{{resolve}}